packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -83,111 +127,151 @@ class UnpackMacOS extends Target {
   }
 }
 
-/// Tell cocoapods to re-fetch dependencies.
-class DebugMacOSPodInstall extends Target {
-  const DebugMacOSPodInstall();
+/// Create an App.framework for debug macOS targets.
+///
+/// This framework needs to exist for the Xcode project to link/bundle,
+/// but it isn't actually executed. To generate something valid, we compile a trivial
+/// string.
+class DebugMacOSFramework extends Target {
+  const DebugMacOSFramework();
+
+  @override
+  String get name => 'debug_macos_framework';
+
+  @override
+  Future<void> build(List<File> inputFiles, Environment environment) async {
+    final FlutterProject flutterProject = FlutterProject.fromDirectory(environment.projectDir);
+    final File outputFile = fs.file(fs.path.join(
+        flutterProject.macos.ephemeralDirectory.path, 'App.framework', 'App'));
+    outputFile.createSync(recursive: true);
+    final File debugApp = environment.buildDir.childFile('debug_app.cc')
+        ..writeAsStringSync(r'''
+static const int Moo = 88;
+''');
+    final RunResult result = await xcode.clang(<String>[
+      '-x',
+      'c',
+      debugApp.path,
+      '-arch', 'x86_64',
+      '-dynamiclib',
+      '-Xlinker', '-rpath', '-Xlinker', '@executable_path/Frameworks',
+      '-Xlinker', '-rpath', '-Xlinker', '@loader_path/Frameworks',
+      '-install_name', '@rpath/App.framework/App',
+      '-o', 'macos/Flutter/ephemeral/App.framework/App',
+    ]);
+    if (result.exitCode != 0) {
+      throw Exception('Failed to compile debug App.framework');
+    }
+  }
 
   @override
-  String get name => 'debug_macos_pod_install';
+  List<Target> get dependencies => const <Target>[];
 
   @override
   List<Source> get inputs => const <Source>[
-    Source.artifact(Artifact.flutterMacOSPodspec,
-      platform: TargetPlatform.darwin_x64,
-      mode: BuildMode.debug
-    ),
-    Source.pattern('{PROJECT_DIR}/macos/Podfile', optional: true),
-    Source.pattern('{PROJECT_DIR}/macos/Runner.xcodeproj/project.pbxproj'),
-    Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/Flutter-Generated.xcconfig'),
+    Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/macos.dart'),
   ];
 
   @override
   List<Source> get outputs => const <Source>[
-    // TODO(jonahwilliams): introduce configuration/planning phase to build.
-    // No outputs because Cocoapods is fully responsible for tracking. plus there
-    // is no concept of an optional output. Instead we will need a build config
-    // phase to conditionally add this rule so that it can be written properly.
+    Source.pattern('{BUILD_DIR}/debug_app.cc'),
```

Comment:
It's this incidental, rather than something we want to treat as an input or an output?

(If we are going to keep it, it seems like the path and filename should be a getter somewhere rather than something you construct twice, with a duplicate string constant and different ways of declaring the path.)

Reply:
I suppose it is actually incidental, given that we don't actual care about its contents